hera_cal/datacontainer.py

```diff
@@ -67,6 +67,16 @@ def __init__(self, data):
                     setattr(self, attr, getattr(data, attr))
                 else:
                     setattr(self, attr, None)
+            
+            # Define the dtype of the data itself. next(iter()) gets the first element
+            # of keys() without creating a whole list.
+            if self.keys():
+                try:
+                    self.dtype = self._data[next(iter(self.keys()))].dtype
```

Comment:
I think this probably fine most of the time, though I wonder whether it makes more sense to to make `.dtype` an `@property`? As it stands, one can init an empty datacontainer with `{}`, and then end up with `self.dtype` as `None`.

Reply:
Ah yeah, I see your point... you could then do a `read` or whatever and the `dtype` wouldn't be updated. I'll make the change.